gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/AbstractDataCompletenessVerifier.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.mapreduce;
+
+import static gobblin.compaction.mapreduce.MRCompactor.FolderState.*;
+
+import gobblin.configuration.State;
+
+
+/**
+ * An abstract class for verifying data completeness of an input folder.
+ *
+ * Subclasses should override the {@link Runnable#run()} method. {@link #afterVerificationStarts()} should be called
+ * right after data completeness verification starts for the input folder, and
+ * {@link #afterVerificationFinishes(boolean)} should be called right after data completeness verification finishes
+ * for the input folder.
+ *
+ * @author ziliu
+ */
+public abstract class AbstractDataCompletenessVerifier
+    implements Runnable, Comparable<AbstractDataCompletenessVerifier> {
+
+  protected final MRCompactor mrCompactor;
+  protected final String inputFolder;
+  protected final State jobProps;
+  protected final double priority;
+
+  public AbstractDataCompletenessVerifier(MRCompactor compactor, String inputFolder, State jobProps, Double priority) {
```

Comment:
`AbstractDataCompletenessVerifier` doesn't contradict with decoupling read and check. It's just an abstract class; its subclasses can first read from kafka audit, then check. Those implementations will be in an RB since they are internal.